control/utils.py

```diff
@@ -182,6 +183,24 @@ def is_valid_nqn(nqn):
 
         return (0, os.strerror(0))
 
+class GatewayUtilsCrypto:
+    ENCRYPTION_KEY = b"Q\xba\x7f._i\x9f%\xed\x96\xe0\xaf\xab k\x9e\xd6\xb6c\xb8\xee\xf6\xa7\xb1s\xf7\xf4\x89f'\x85b"
```

Comment:
rgw's server-side encryption relies on a key management server (vault mostly). we don't store any key material in rados

Reply:
@cbodley does the key server offer more security? Who can access the keys stored there? I assume that the access is more restricted than the one for the OMAP content?

Reply:
Vault seems like an excellent tool for managing secrets and keys. In general, there can be multiple key storage backends, The sibling project, NooBaa Operator, for example, offers a framework to support multiple [KMS backends](https://github.com/noobaa/noobaa-operator/tree/master/pkg/util/kms) 😀

Reply:
We now get the encryption key from the user, I kept a hardcoded one just as a fallback in case the user didn't want to define his own key. In this case we can either do nothing at all and store the keys unencrypted or still use some encryption.

Reply:
@epuertat Built-in encryption in librbd basically mimics LUKS (to the point that it's fully compatible with e.g. dm-crypt for standalone images).  The key material is stored in RADOS, but it's wrapped the same way as it would be if the user employed dm-crypt on top of a "dumb", non-encrypted RBD device.  None of that feels applicable here.

Reply:
Thank you, @idryomov and @cbodley for the info!